src/nemo_safe_synthesizer/data_processing/records/value_path.py

```
@@ -96,18 +96,20 @@ def unflatten(data: dict[ValuePath, Any]) -> Optional[dict | list]:
     return result
 
 
-def _ensure_array_size(result: list, item: int):
+def _ensure_array_size(result: list[Any], item: int) -> None:
     if len(result) <= item:
         for i in range(len(result), item + 1):
```

Ruff (0.15.20)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)